Src/Support/Google.Apis.Auth.Tests/OAuth2/ImpersonatedCredentialTests.cs

```diff
@@ -0,0 +1,230 @@
+            Assert.Throws<ArgumentException>(() => new ImpersonatedCredential(initializer));
+        }
+
+        [Fact]
+        public void TestConstructorInvalidLifetime()
+        {
+            var initializer = new ImpersonatedCredential.Initializer(CreateSourceCredential(), "foo", null, null, 50000);
+            var ex = Assert.Throws<InvalidOperationException>(() => new ImpersonatedCredential(initializer));
+            Assert.Equal("Lifetime must be less than or equal to 43200.", ex.Message);
+        }
+
+        [Fact]
+        public void TestConstructor()
+        {
+            var delegates = new string[] {"delegate"};
+            var scopes = new string[] {"scope"};
+            var initializer = new ImpersonatedCredential.Initializer(CreateSourceCredential(), "foo", delegates, scopes);
+            var credential = new ImpersonatedCredential(initializer);
+            Assert.Equal(3600, credential.LifetimeInSeconds);
+            Assert.Equal(delegates, credential.Delegates);
+            Assert.Equal(scopes, credential.Scopes);
+            Assert.Equal("foo", credential.TargetPrincipal);
+        }
+
+        [Fact]
+        public async Task TestFetchesOidcToken()
+        {
+            var credential = CreateImpersonatedCredentialWithIdTokenResponse();
+            var oidcToken = await credential.GetOidcTokenAsync(OidcTokenOptions.FromTargetAudience("will.be.ignored"));
+            var token = await oidcToken.GetAccessTokenAsync();
+            Assert.Equal(674146, oidcToken.TokenResponse.ExpiresInSeconds);
+            Assert.Equal(OidcComputeSuccessMessageHandler.FirstCallToken, token);
+        }
+
+        [Fact]
+        public async Task TestFetchesOidcTokenFailure()
+        {
+            var credential = CreateImpersonatedCredentialWithErrorResponse();
+            var oidcToken = await credential.GetOidcTokenAsync(OidcTokenOptions.FromTargetAudience("will.be.ignored"));
+            var ex = await Assert.ThrowsAsync<TokenResponseException>(async () =>{var result = await oidcToken.GetAccessTokenAsync();});
+            Assert.Equal(errorResponseContent, ex.Error.Error);
+        }
+
+        [Fact]
+        public async Task TestFetchesAccessToken()
+        {
+            var credential = CreateImpersonatedCredentialWithAccessTokenResponse();
+            var success = await credential.RequestAccessTokenAsync(CancellationToken.None).ConfigureAwait(false);
+            Assert.True(success);
+            Assert.Equal(3600, credential.Token.ExpiresInSeconds);
+            Assert.Equal("access_token", credential.Token.AccessToken);
+        }
+
+        [Fact]
+        public async Task TestFetchesAccessTokenFailure()
+        {
+            var credential = CreateImpersonatedCredentialWithErrorResponse();
+            var ex = await Assert.ThrowsAsync<TokenResponseException>(async () =>{await credential.RequestAccessTokenAsync(CancellationToken.None).ConfigureAwait(false);});
```

Comment:
See comment for TestSignBlobFailure - this can be simplified.

Reply:
done